middleware/basic_auth.go

```diff
@@ -93,7 +93,7 @@ func BasicAuthWithConfig(config BasicAuthConfig) echo.MiddlewareFunc {
 				}
 			}
 
-			realm := ""
+			var realm string
```

Comment:
As above

Reply:
ineffectual assignment

Reply:
How about this?

```go
realm := defaultRealm

if config.Realm != defaultRealm {
    realm = strconv.Quote(config.Realm)
}
```

or maybe have defaultRealm unquoted and the entire variable and if can go away, and on the response we can simply use `strconv.Quote(config.Realm)` directly.

Reply:
thanks.. fixed